code/ship/ship.cpp

```diff
@@ -19088,3 +19123,162 @@ bool ship::is_arriving(ship::warpstage stage, bool dock_leader_or_single)
 	Assertion(false, "ship::is_arriving didn't handle all possible states; get a coder!");
 	return false;
 }
+
+void ship_clear_lock(lock_info *slot) {
+	vec3d zero_vec = ZERO_VECTOR;
+
+	slot->accumulated_x_pixels = 0;
+	slot->accumulated_y_pixels = 0;
+
+	slot->catch_up_distance = 0.0f;
+
+	slot->catching_up = 0;
+
+	slot->current_target_sx = -1;
+	slot->current_target_sy = -1;
+
+	slot->dist_to_lock = 0.0f;
+
+	slot->indicator_start_x = -1;
+	slot->indicator_start_y = -1;
+
+	slot->indicator_visible = false;
+
+	slot->indicator_x = -1;
+	slot->indicator_y = -1;
+
+	slot->last_dist_to_target = 0.0f;
+
+	slot->locked_timestamp = 0;
+
+	slot->maintain_lock_count = 0;
+
+	slot->need_new_start_pos = 0;
+
+	slot->target_in_lock_cone = false;
+
+	slot->world_pos = zero_vec;
+
+	slot->locked = false;
+
+	slot->obj = NULL;
+	slot->subsys = NULL;
+
+
+	slot->time_to_lock = -1.0f;
+}
+
+void ship_queue_missile_locks(ship *shipp, weapon_info *wip)
+{
+	size_t i;
+
+	shipp->missile_locks_firing.clear();
+
+	// queue up valid missile locks
+	for ( i = 0; i < shipp->missile_locks.size(); ++i ) {
+		if ( shipp->missile_locks[i].locked ) {
+			shipp->missile_locks_firing.push_back(shipp->missile_locks[i]);
+		}
+	}
+}
+
+bool ship_lock_present(ship *shipp)
+{
+	size_t i;
+
+	for ( i = 0; i < shipp->missile_locks.size(); ++i ) {
```

Comment:
this loop could be replaced with a call to ```any_of(std::begin(ship->missile_locks), std::end(ship->missile_locks), [](auto it) {return it->locked;});```

(possible syntax errors, since I didn't test it)

Reply:
I believe this should work:
```cpp
return any_of(shipp->missile_lock.begin(), shipp->missile_lock.end(), 
           [](lock_info lock){ return lock.locked; });
```

Reply:
looks good, just a small typo. ´´missile_locks´´ with an s